### PR TITLE
vyper/functions/signature.py lint

### PR DESCRIPTION
**Pull request by davesque**
Foregoing the animal pictures and descriptions with these lint PRs.  See #1262 and #1276 .

### CODE REVIEW

File: vyper/functions/signature.py [hunk 1]
```diff
@@ -69,17 +72,30 @@ def process_arg(index, arg, expected_arg_typelist, function_name, context):
...
-                elif expected_arg in ('int128', 'uint256') and \
-                     isinstance(vsub.typ, BaseType) and \
-                     vsub.typ.typ in ('int128', 'uint256') and \
-                     vsub.typ.is_literal and \
-                     SizeLimits.in_bounds(expected_arg, vsub.value):
+                elif is_valid_integer:
                     return vsub
+                # TODO: missing final else clasue allowing implicit exit of
+                # this block without hitting any of the logic.  Is this
+                # intended?
```
Comment (fubuloubu) [inline vyper/functions/signature.py]: @jacqueswww, should evaluate this

Comment (jacqueswww) [reply vyper/functions/signature.py]: @davesque remove this TODO, if anything fails the exception below will pick it up.

Comment (jacqueswww) [reply vyper/functions/signature.py]: @davesque I believe `return arg.id` on line 109 is unreachable code as well.

**Comment by davesque** [comment]
@jacqueswww Addressed both comments and rebased.  Should be good to merge.

**Pull request merged**